### PR TITLE
Features/lease preference used #157669475

### PR DESCRIPTION
**Pull request by akegan**
https://www.pivotaltracker.com/story/show/157669475

### CODE REVIEW

File: app/javascript/components/supplemental_application/sections/LeaseInformatonInputs.js
Comment (fddayan) [inline app/javascript/components/supplemental_application/sections/LeaseInformatonInputs.js]: application can be unpacked in line 37

File: app/javascript/components/supplemental_application/SupplementalApplicationPage.js
Comment (akegan) [inline app/javascript/components/supplemental_application/SupplementalApplicationPage.js]: try using null as the id instead of 'no_preference_used'

Comment (akegan) [reply app/javascript/components/supplemental_application/SupplementalApplicationPage.js]: doesn't work

File: app/javascript/components/supplemental_application/sections/LeaseInformatonInputs.js
Comment (akegan) [inline app/javascript/components/supplemental_application/sections/LeaseInformatonInputs.js]: look into a hidden field

File: app/javascript/components/supplemental_application/sections/LeaseInformatonInputs.js
Comment (chrisdolendo) [inline app/javascript/components/supplemental_application/sections/LeaseInformatonInputs.js]: @akegan created an onchange function so that the value assignment for `noPreferenceUsed` stays within this file :)

File: app/javascript/components/supplemental_application/sections/LeaseInformatonInputs.js
Comment (chrisdolendo) [inline app/javascript/components/supplemental_application/sections/LeaseInformatonInputs.js]: I just changed the `no_preferenced_used` string to `null` so that the value doesn't need to be changed to null on submit. I would vote to change the name of 'id' to 'value', but the functionality still works either way.

File: app/javascript/components/supplemental_application/SupplementalApplicationPage.js
Comment (chrisdolendo) [inline app/javascript/components/supplemental_application/SupplementalApplicationPage.js]: I initially moved this to the `LeaseInformationInputs` component but kept it here so it stays centralized to this function.

File: app/services/force/lease_service.rb
Comment (chrisdolendo) [inline app/services/force/lease_service.rb:34]: A little refactor here and on update so we can clearly see the difference in the params.

File: app/javascript/components/supplemental_application/SupplementalApplicationPage.js
Comment (chrisdolendo) [inline app/javascript/components/supplemental_application/SupplementalApplicationPage.js:234]: The dropdown for `preferenceUsed` works such as:
If the value of `preference_used` is `null`, the dropdown will show "None".
If there is no `preference_used` key in the lease object, the dropdown will show the placeholder "Select One".

Therefore, I created this logic (@akegan this was based on our discussion) to delete the key, and know for certain that `preference_used` was an untouched field when first loading the page.

File: app/services/force/lease_service.rb
Comment (akegan) [inline app/services/force/lease_service.rb]: we need the lease status in here too

**Comment by akegan** [comment]
I removed console logs and updated the snapshots. Looks good to me except for the comment above about including the lease status on update. 

I think that the FIXME about wondering how it will work with the preference update is still valid, and we won't really be able to know for sure until @callaghanc's PR gets merged. I'm up for creating a followup ticket or waiting til it's ready for QA and then rebasing on top. Either way.

**Comment by chrisdolendo** [comment]
@akegan added my update here, and handing back off to you. Let me know if you need to pair.

**Comment by akegan** [comment]
Update: currently the dropdown does not update based on status updates on the preferences being saved. I need to look into what is updating and what is not updating and possible pair with @fddayan tomorrow morning to discuss

**Comment by akegan** [comment]
Blocked by this interaction pr: https://github.com/Exygy/sf-dahlia-lap/pull/104 

(That PR will fix the issue that the dropdown options are not updated when preferences are confirmed)

**Comment by callaghanc** [comment]
@akegan We've just merged the confirmed prefs panel UX behavior PR, so this PR is now unblocked!

**Comment by akegan** [comment]
Testing this now to see if it works as-is

**Comment by akegan** [comment]
It is very close to working, but not quite. the available options in the dropdown only change if you change the selection, then the next time you click on it you get the updated options 😕. looking into it more now

**Comment by akegan** [comment]
This is ready for review. I'm working on adding tests, but I'm wondering if we should get this an initial dev review and then we can start on product acceptance while I work on figuring out how to add some unit tests

File: app/javascript/components/supplemental_application/sections/LeaseInformationInputs.js
Comment (fddayan) [inline app/javascript/components/supplemental_application/sections/LeaseInformationInputs.js]: What is the intention of this? 
`toggleNoPreferenceUsed` in line 36 should be returning a function that can be used as a the event handler for `onChange`. Any particular reason we are doing this?

File: app/javascript/components/supplemental_application/sections/LeaseInformationInputs.js
Comment (fddayan) [inline app/javascript/components/supplemental_application/sections/LeaseInformationInputs.js]: values should not be modified directly. Instead `formApi.setValue(field, value)` should be used.
react-form refs for our version: https://dazzling-keller-a8107a.netlify.com/#/

Reading from `formApi.value` is ok 👍

File: app/javascript/components/supplemental_application/sections/LeaseInformationInputs.js
Comment (chrisdolendo) [reply app/javascript/components/supplemental_application/sections/LeaseInformationInputs.js]: @fddayan I wrote this code so responding -- So you're saying it should be `onChange={toggleNoPreferenceUsed}` instead of `onChange={toggleNoPreferenceUsed(formApi)}`, I remember I tried using the former (`{toggleNoPreferenceUsed}`) but it wasn't working 100% correctly.

File: app/javascript/components/supplemental_application/sections/LeaseInformationInputs.js
Comment (chrisdolendo) [reply app/javascript/components/supplemental_application/sections/LeaseInformationInputs.js]: @fddayan Good catch, I will change to use `setValue` instead.

**Review by fddayan — APPROVED** [review]
looks good 👍

File: app/javascript/components/supplemental_application/sections/LeaseInformationInputs.js
Comment (chrisdolendo) [inline app/javascript/components/supplemental_application/sections/LeaseInformationInputs.js:61]: @fddayan 😄

**Pull request merged**
